src/oop/user_defined_classes/hard/q3/MainTest.java

```diff
@@ -0,0 +1,142 @@
+        String testClassString = "oop.user_defined_classes.hard.q3." + testClassName;
+        classInstance = new ObjectTest(testClassString);
+    }
+
+    @Test
+    public void cupboardClassHasCorrectAttributes() {
+        String missingAttributeMessage =
+                "The %s attribute could not be found in your %s class. Please make sure you have added it and that it is spelled correctly.";
+        String wrongTypeMessage = "The %s attribute does not have the correct type.";
+        String incorrectModifierMessage = "Your %s attribute in the %s class does not have the correct visibility modifier.";
+        assertTrue(classInstance.hasField(intAttributeName1), String.format(missingAttributeMessage, intAttributeName1, testClassName));
+        assertTrue(classInstance.hasField(intAttributeName1, int.class),
+                String.format(wrongTypeMessage, intAttributeName1));
+        assertTrue(classInstance.hasModifier(intAttributeName1, "private"),
+                String.format(incorrectModifierMessage, intAttributeName1, testClassName));
+        assertTrue(classInstance.hasField(booleanAttributeName1), String.format(missingAttributeMessage, booleanAttributeName1, testClassName));
+        assertTrue(classInstance.hasField(booleanAttributeName1, boolean.class),
+                String.format(wrongTypeMessage, booleanAttributeName1));
+        assertTrue(classInstance.hasModifier(booleanAttributeName1, "private"),
+                String.format(incorrectModifierMessage, booleanAttributeName1, testClassName));
+    }
+
+    @Test
+    public void cupboardClassHasFirstRequiredConstructor() {
+        assertTrue(classInstance.hasConstructor(null),
+                String.format("Your %s class is missing the parameterless required constructor.", testClassName));
+        assertTrue(classInstance.hasConstructor(null, new String[]{"public"}),
+                String.format("Your parameterless %s class constructor has the incorrect visibility modifier.", testClassName));
+    }
+
+    @Test
+    public void cupboardClassHasSecondRequiredConstructor() {
+        assertTrue(classInstance.hasConstructor(null),
+                String.format("Your %s class is missing the required constructor with parameters.", testClassName));
+        assertTrue(classInstance.hasConstructor(null, new String[]{"public"}),
+                String.format("Your %s class constructor with parameters has the incorrect visibility modifier.", testClassName));
+    }
+
+    @Test
+    public void cupboardFirstConstructorInitializesAttributesCorrectly() throws Throwable {
+        Object instance = classInstance.createInstance();
+        String incorrectAttributeInstantiationMessage =
+                "Your " + testClassName + " parameterless constructor does not correctly initialize the object.";
+        _assertEquals(startPotsValue, classInstance.getFieldValue(instance, intAttributeName1),
+                incorrectAttributeInstantiationMessage);
```

Comment:
You can also change these error messages to say which attribute they did not initialize correctly.